src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -722,24 +800,24 @@ protected void runSpooler() {
     }
 
     @SuppressWarnings("PMD.CloseResource")
-    private synchronized AwsIotMqttClient getConnection(boolean forSubscription) {
+    private synchronized IndividualMqttClient getConnection(boolean forSubscription) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.